lib/elixir_tic_tac_toe_basic/ui.ex

```diff
@@ -3,13 +3,22 @@ defmodule ElixirTicTacToeBasic.UI do
   Provides a user interface.
   """
 
-  def welcome(state, callback \\ fn m -> IO.puts(m) end) do
+  def welcome(
+        state,
+        callback \\ fn message ->
+          IO.puts(message)
+          message
+        end
```

Comment:
**suggestion**: I believe you can use a module attribute remove the duplication between these two functions.

```elixir
@callback fn message -> IO.puts(message) && message end

def welcome(state, callback \\ @callback) do
```

Reply:
I'll come back to this one. 

Notes to self for later:

When I use `@callback`, I get an error that the type specification is missing the return type. I think `@callback `is a reserved word to specify required functions for modules that adopt a given behaviour. 

When I changed it to `@message`, it gave me this argument error: 

`(ArgumentError) cannot inject attribute @message into function/macro because cannot escape #Function<0.115126747 in file:lib/elixir_tic_tac_toe_basic/ui.ex>. The supported values are: lists, tuples, maps, atoms, numbers, bitstrings, PIDs and remote functions in the format &Mod.fun/arity`

Reply:
I think `@callback` might be reserved for Behaviours.